File: DsWorkJava/src/testwork/controller/demo/DemoController.java
/**
 * 功能:DEMOController
 * 开发人员:你是谁
 * 创建时间:2013-9-13 23:06:34
 */
package testwork.controller.demo;

import java.io.PrintWriter;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import dswork.web.MyRequest;
import dswork.core.page.Page;
import dswork.core.page.PageNav;
import dswork.core.page.PageRequest;
import dswork.core.util.CollectionUtil;
import testwork.model.Demo;
import testwork.service.demo.DemoService;

@Controller
@RequestMapping("/manage/demo")
public class DemoController
{
	@Autowired
	private DemoService service;

	@RequestMapping("/addDemo1")
	public String addDemo1()
	{
		return "/view/manage/demo/addDemo.jsp";
	}
	
	@RequestMapping("/addDemo2")
	public void addDemo2(PrintWriter out, Demo po)
	{
		try
		{
			service.save(po);
			out.print(1);
		}
		catch (Exception e)
		{
			e.printStackTrace();
			out.print("0:" + e.getMessage());
		}
	}
	
	@RequestMapping("/delDemo")
	public void delDemo(HttpServletRequest request, HttpServletResponse response)
	{
		MyRequest req = new MyRequest(request);
		service.deleteBatch(CollectionUtil.toLongArray(req.getLongArray("keyIndex", 0)));
		try
		{
			response.sendRedirect(req.getRefererURL());
		}
		catch(Exception ex)
		{
		}
	}
	
	@RequestMapping("/updDemo1")
	public String updDemo1(HttpServletRequest request)
	{
		MyRequest req = new MyRequest(request);
		Long id = req.getLong("keyIndex");
		request.setAttribute("po", service.get(id));
		request.setAttribute("page", req.getInt("page", 1));
		return "/view/manage/demo/updDemo.jsp";
	}
	
	@RequestMapping("/updDemo2")
	public void updDemo2(PrintWriter out, Demo po)
	{
		try
		{
			service.update(po);
			out.print(1);
		}
		catch (Exception e)
		{
			e.printStackTrace();
			out.print("0:" + e.getMessage());
		}
	}
	
	@RequestMapping("/getDemo")
	public String getDemo(HttpServletRequest request)
	{
		MyRequest req = new MyRequest(request);
		PageRequest pr = new PageRequest();
		pr.setFilters(req.getParameterValueMap(false, false));
		pr.setCurrentPage(req.getInt("page", 1));
		pr.setPageSize(req.getInt("pageSize", 10));
		Page<Demo> pageModel = service.queryPage(pr);
		request.setAttribute("pageModel", pageModel);
		request.setAttribute("pageNav", new PageNav<Demo>(request, pageModel));
		return "/view/manage/demo/getDemo.jsp";
	}
	
	@RequestMapping("/getDemoById")
	public String getDemoById(HttpServletRequest request)
	{
		MyRequest req = new MyRequest(request);
		Long id = req.getLong("keyIndex");
		request.setAttribute("po", service.get(id));
		return "/view/manage/demo/getDemoById.jsp";
	}
}

File: DsDemo/DemoNet/DemoNet/Areas/Manage/Controllers/DemoController.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Web;
using System.Web.Mvc;

using Dswork.Core.Page;
using Dswork.Web;
using Dswork.Mvc;

using DemoNet.Model;
using DemoNet.Service;

namespace Areas.Manage.Controllers
{
	public class DemoController:BaseController
    {
		public virtual DemoService demoService { set; protected get; }

        public ActionResult AddDemo1()
		{
            return View();
        }

		[HttpPost]
		public String AddDemo2(Demo po)
        {
			try
			{
                demoService.Save(po);
				return "1";
			}
			catch (Exception e)
			{
				return "0:" + e.Message;
			}
        }

		[AcceptVerbs(HttpVerbs.Post|HttpVerbs.Get)]
		public String DelDemo()
		{
			try
			{
			  demoService.DeleteBatch(req.GetLongArray("keyIndex", 0));
			  return "1";
			}
			catch (Exception e)
			{
				return "0:" + e.Message;
			}
		}

		public ActionResult UpdDemo1(long keyIndex)
		{
            ViewBag.po = demoService.Get(keyIndex);
			ViewBag.page = req.GetInt("page", 1);
			return View();
		}

		[HttpPost]
		public String UpdDemo2(Demo po)
		{
			try
			{
                demoService.Update(po);
				return "1";
			}
			catch(Exception e)
			{
				return "0:" + e.Message;
			}
		}

		[AcceptVerbs(HttpVerbs.Post | HttpVerbs.Get)]
		public ActionResult GetDemo()
		{
			PageRequest rq = GetPageRequest();
			rq.Filters = req.GetParameterValueMap(false, false);
            Page<Demo> pageModel = demoService.QueryPage(rq);
			ViewBag.pageModel = pageModel;
			ViewBag.result = pageModel.GetResult<Demo>();
			ViewBag.pageNav = (new PageNav<Demo>(Request, pageModel));
			return View();
		}

		public ActionResult GetDemoById(long keyIndex)
		{
            ViewBag.po = demoService.Get(keyIndex);
			return View();
        }
    }
}
